pkg/services/datasources/service/datasource_test.go

```diff
@@ -187,6 +187,92 @@ func TestService_UpdateDataSource(t *testing.T) {
 		_, err = dsService.UpdateDataSource(context.Background(), cmd)
 		require.ErrorIs(t, err, datasources.ErrDataSourceNameExists)
 	})
+
+	t.Run("should merge cmd.SecureJsonData with db data", func(t *testing.T) {
+		sqlStore := db.InitTestDB(t)
+		secretsService := secretsmng.SetupTestService(t, fakes.NewFakeSecretsStore())
+		secretsStore := secretskvs.NewSQLSecretsKVStore(sqlStore, secretsService, log.New("test.logger"))
+		quotaService := quotatest.New(false, nil)
+		mockPermission := acmock.NewMockedPermissionsService()
+		dsService, err := ProvideService(sqlStore, secretsService, secretsStore, cfg, featuremgmt.WithFeatures(), actest.FakeAccessControl{}, mockPermission, quotaService)
+		require.NoError(t, err)
+
+		mockPermission.On("SetPermissions", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return([]accesscontrol.ResourcePermission{}, nil)
+
+		expectedDbKey := "db-secure-key"
+		expectedDbValue := "db-secure-value"
+		ds, err := dsService.AddDataSource(context.Background(), &datasources.AddDataSourceCommand{
+			OrgID: 1,
+			Name:  "test-datasource",
+			SecureJsonData: map[string]string{
+				expectedDbKey: expectedDbValue,
+			},
+		})
+		require.NoError(t, err)
+
+		expectedOgKey := "cmd-secure-key"
+		expectedOgValue := "cmd-secure-value"
+
+		cmd := &datasources.UpdateDataSourceCommand{
+			ID:    ds.ID,
+			OrgID: ds.OrgID,
+			Name:  "test-datasource-updated",
+			SecureJsonData: map[string]string{
+				expectedOgKey: expectedOgValue,
+			},
+		}
+
+		_, err = dsService.UpdateDataSource(context.Background(), cmd)
+		require.NoError(t, err)
+
+		assert.Equal(t, cmd.SecureJsonData[expectedDbKey], expectedDbValue)
+		assert.Equal(t, cmd.SecureJsonData[expectedOgKey], expectedOgValue)
```

Comment:
I am confused by those assertions, which are run on `cmd` which is the `UpdateDataSourceCommand` (so the input).

 Shouldn't they be like this:

```suggestion
		ds, err = dsService.UpdateDataSource(context.Background(), cmd)
		require.NoError(t, err)

		secret, err := dsService.DecryptedValues(context.Background(), ds)
		require.NoError(t, err)

		assert.Equal(t, secret[expectedDbKey], expectedDbValue)
		assert.Equal(t, secret[expectedOgKey], expectedOgValue)
```

Reply:
I wanted to test that `fillWithSecureJSONData` was indeed altering the `cmd` and as the `cmd` is a pointer argument these checks are possible in the caller context.
But you are right, checking the `ds` also shows that everything works indirectly, because this is the return from the db update.
I'll to the changes.

Reply:
Ah yes, makes sense! Thanks for addressing anyways! :raised_hands: